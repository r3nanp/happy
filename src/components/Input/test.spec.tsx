import { render, screen, waitFor } from 'utils/tests/test-utils'
import userEvent from '@testing-library/user-event'
import { Input } from '.'

describe('<Input />', () => {
  it('Should be able to render a input component with label', () => {
    render(<Input label="Label" name="Label" />)

    expect(screen.getByLabelText('Label')).toBeInTheDocument()
  })

  it('Should be able to render without label', () => {
    render(<Input />)

    expect(screen.queryByLabelText('Label')).not.toBeInTheDocument()
  })

  it('Should be able to render with a placeholder', () => {
    render(<Input placeholder="testing" />)

    expect(screen.getByPlaceholderText('testing')).toBeInTheDocument()
  })

  it('Changes its value when typing', async () => {
    const onInputChange = jest.fn()
    render(<Input onInputChange={onInputChange} label="Input" name="Input" />)

    const input = screen.getByRole('textbox')
    const text = 'This is my new text'
    userEvent.type(input, text)

    await waitFor(() => {
      expect(input).toHaveValue(text)
      expect(onInputChange).toHaveBeenCalledTimes(text.length)
    })
    expect(onInputChange).toHaveBeenCalledWith(text)
  })

  it('Does not changes its value when disabled', async () => {
    const onInputChange = jest.fn()
    render(
      <Input
        onInputChange={onInputChange}
        label="Input"
        name="Input"
        disabled
      />
    )

    const input = screen.getByRole('textbox')
    expect(input).toBeDisabled()

    const text = 'This is my new text'
    userEvent.type(input, text)

    await waitFor(() => {
      expect(input).not.toHaveValue(text)
    })
    expect(onInputChange).not.toHaveBeenCalled()
  })

  it('Renders with error', () => {
    const { container } = render(<Input label="Error" error="Error message" />)

    expect(screen.getByText('Error message')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
