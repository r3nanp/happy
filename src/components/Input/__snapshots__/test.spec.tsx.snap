// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Renders with error 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #8FA7B3;
  font-weight: 800;
  margin-bottom: 0.5rem;
  line-height: 1.5rem;
}

<label
  class="c0"
>
  Error
</label>
`;
